python/unitytrainers/ppo/trainer.py

```diff
@@ -232,7 +232,7 @@ def generate_value_estimate(self, brain_info, idx):
         feed_dict = {self.model.batch_size: 1, self.model.sequence_length: 1}
         if self.use_visual_obs:
             for i in range(len(brain_info.visual_observations)):
-                feed_dict[self.model.visual_in[i]] = brain_info.visual_observations[i][idx]
+                feed_dict[self.model.visual_in[i]] = [brain_info.visual_observations[i][idx]]
```

Comment:
Why is this a list ? The fact that this works seems counter intuitive.

Reply:
TensorFlow accepts a list as the batch dimension.

Reply:
Well if it works... Was that tested with multiple observations ?

Reply:
I just tested it with the GridWorld environment on LInux. What do you mean multiple observations?

Reply:
An environment with more than one camera. Try with the tester environment.

Reply:
@vincentpierre I tried this branch with a linux executable built using your Tester branch, and it gave me this error. 

UnityAgentsException: The brain TesterBasicDCDOBrain was set to External mode but Unity was unable to read the arguments passed at launch.
  at CoreBrainExternal.InitializeCoreBrain (Communicator communicator) [0x0002c] in <d2ff26f93b7c430ea2967fdd4f468b4f>:0
  at Brain.InitializeBrain (Academy aca, Communicator communicator) [0x00006] in <d2ff26f93b7c430ea2967fdd4f468b4f>:0
  at Academy.InitializeEnvironment () [0x00049] in <d2ff26f93b7c430ea2967fdd4f468b4f>:0
  at Academy.Awake () [0x00000] in <d2ff26f93b7c430ea2967fdd4f468b4f>:0

Any idea why this happens?